controlplane/kubeadm/internal/filters.go

```diff
@@ -41,65 +40,73 @@ import (
 // - mutated in-place (ex: NodeDrainTimeout)
 // - are not dictated by KCP (ex: ProviderID)
 // - are not relevant for the rollout decision (ex: failureDomain).
-func matchesMachineSpec(infraConfigs map[string]*unstructured.Unstructured, machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane, machine *clusterv1.Machine) (string, bool, error) {
-	mismatchReasons := []string{}
+func matchesMachineSpec(infraConfigs map[string]*unstructured.Unstructured, machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane, machine *clusterv1.Machine) (bool, []string, []string, error) {
+	logReasons := []string{}
+	conditionReasons := []string{}
```

Comment:
```suggestion
	mismatchLogReasons := []string{}
	mismatchConditionReasons := []string{}
```

To not lose the semantic of what these reasons are about (probably similar below)

Reply:
I replaced Reasons with Messages, to avoid confusion with condition reasons.
I would prefer to not add mismatch, it seems redoundant